source/Hardpoint.cpp

```diff
@@ -178,58 +178,66 @@ void Hardpoint::Step()
 		return;
 
 	wasFiring = isFiring;
-	if(reload > 0.)
-		--reload;
-	// If the full reload time is elapsed, reset the burst counter.
-	if(reload <= 0.)
-		burstCount = outfit->BurstCount();
-	// If the weapon recently fired then increment any sustained fire counters.
 	if(reload > 0.)
 	{
+		--reload;
+		// If the weapon recently fired then increment any sustained fire counters.
 		if(bloomCount < outfit->BloomTime())
 			++bloomCount;
 		if(spinupCount < outfit->SpinupTime())
 			++spinupCount;
 	}
-	// Only decrement sustained fire counters if this weapon could have fired
-	// but didn't.
-	else if(!wasFiring)
+	if(reload <= 0.)
 	{
-		if(bloomCount > 0.)
-			--bloomCount;
-		if(spinupCount > 0.)
-			--spinupCount;
+		// If the full reload time is elapsed, reset the burst counter.
+		burstCount = outfit->BurstCount();
+
+		// Only decrement sustained fire counters if this weapon could have fired
+		// but didn't.
+		if(!wasFiring)
+		{
+			if(bloomCount > 0.)
+				--bloomCount;
+			if(spinupCount > 0.)
+				--spinupCount;
+		}
 	}
 
 	if(burstReload > 0.)
-		--burstReload;
-	// If the burst reload time has elapsed, this weapon will not count as firing
-	// continuously if it is not fired this frame.
-	if(burstReload <= 0.)
-		isFiring = false;
-	// If a burst has run out then reset the burst sustained fire counters.
-	if(burstCount == 0)
 	{
-		burstBloomCount = 0.;
-		burstSpinupCount = 0.;
-	}
-	// If the weapon recently fired a burst then increment any burst
-	// sustained fire counters.
-	else if(burstReload > 0.)
-	{
-		if(burstBloomCount < outfit->BurstBloomTime())
-			++burstBloomCount;
-		if(burstSpinupCount < outfit->BurstSpinupTime())
-			++burstSpinupCount;
+		--burstReload;
+
+		// If a burst has run out then reset the burst sustained fire counters.
+		if(burstCount == 0)
+		{
+			burstBloomCount = 0.;
+			burstSpinupCount = 0.;
+		}
+		else
+		{
+			// If the weapon recently fired a burst then increment any burst
+			// sustained fire counters.
+			if(burstBloomCount < outfit->BurstBloomTime())
+				++burstBloomCount;
+			if(burstSpinupCount < outfit->BurstSpinupTime())
+				++burstSpinupCount;
+		}
 	}
-	// Only decrement burst sustained fire counters if this weapon could
-	// have fired but didn't.
-	else if(!wasFiring)
+	if(burstReload <= 0.)
```

Comment:
Same here.
```suggestion
	else
```

Reply:
Because the contents of these rely on the state of reload and burstReload *after* it has been decremented.

Reply:
Ok yeah you're right.